diff --git a/src/Pact/Types/Util.hs b/src/Pact/Types/Util.hs
index 0bd22cf4..463c8c1b 100644
--- a/src/Pact/Types/Util.hs
+++ b/src/Pact/Types/Util.hs
@@ -148,8 +148,8 @@ parseB16JSON = withText "Base16" parseB16Text
 
 parseB16Text :: Text -> Parser ByteString
 parseB16Text t = case B16.decode (encodeUtf8 t) of
-                 (s,leftovers) | leftovers == B.empty -> return s
-                               | otherwise -> fail $ "Base16 decode failed: " ++ show t
+                 Right s -> return s
+                 Left _ -> fail $ "Base16 decode failed: " ++ show t
 {-# INLINE parseB16Text #-}
 
 parseB16TextOnly :: Text -> Either String ByteString
